src/nsntrace.c

```diff
@@ -205,8 +205,12 @@ _nsntrace_start_tracee(struct nsntrace_options *options)
 		uid = getuid();
 		gid = getgid();
 	}
-	setgid(gid);
-	setuid(uid);
+	if (setgid(gid) < 0) {
+		fprintf(stderr, "Unable to set process GID");
+	}
+	if (setuid(uid) < 0) {
+		fprintf(stderr, "Unable to set process UID");
+	}
```

Comment:
The question is, should the program still be executed with too many rights, or not.
I don't know.